Modem (#6)
hostmobility/documentation: docs/interfaces/c61/modem.md

---
title: C61 Modem
tags:
    - C61

>     - CT GL

Reviewer: I think it's strange to list CT GL when the title is C61

Author: Alternative is to copy all to CT GL folder?

Reviewer: OK, if want :-)

---

## Overview 

The C61 has a modem from Quectel called EG25-G with a GPS receiver.

## Starting/Enabling

**Note:**  The modem is turned off by default.
```
echo 1 > /opt/hm/pic_attributes/ctrl_modem_on
```

> * The modem takes about 10 seconds to start and register itself with the kernel. And thakes up to one minutes before it is registered in the linux usb list.

Reviewer: ```suggestion
* The modem takes about 10 seconds to start and register itself with the kernel. It then takes up to one minute before it is registered in the linux usb list.
```

* Check USB TTY device to see if it has shown up.

```bash
find /dev -name "ttyUSB*"
```

```bash
/dev/ttyUSB3
/dev/ttyUSB2
/dev/ttyUSB1
/dev/ttyUSB0
```

* The modem also shows up with the `lsusb` command.

```bash
root@10210100023:~# lsusb | grep Quectel
Bus 001 Device 007: ID 2c7c:030a Quectel Wireless Solutions Co., Ltd. Quectel EM05-G
```

> # Status ON

Reviewer: ```suggestion
# Status (ON/OFF)
```

Author: it is the status on status off is not showing as an example

Reviewer: OK, good. Then maybe clarify that

```bash
/opt/hm/modem_status.sh
******* MODEM STATUS DUMP *******

ctrl_modem_status: 1342177491

Modem state     ON
Ctrl state:     ON

SLED:           0
SLED Mon:       ON

Modem EMG/reset:        0

Current state:  on
Prev state:     turn_on_timeout

Prev RC:        unexpected_shutdown

Start cnt:      0
Reset cnt:      0

EMG used:       NO
```

## Shutdown/Reset 

> To disable it, set ctrl_modem_on to 0
> 
> To reset it use ctrl_modem_emg and set it to 0 then back to 1 after a second.

Reviewer: ```suggestion
To disable the modem, set ctrl_modem_on to 0

To reset it use ctrl_modem_emg and set it to 0, then back to 1 after a second.
```

```bash
echo 0 > /opt/hm/pic_attributes/ctrl_modem_on
# or (not to be used regulary)

> echo 0 > /opt/hm/pic_attributes/ctrl_modem_emg

Reviewer: plz clarify

Author: I have read that in gemalto datasheet that it is not recommed to use the emergeny shutdown if you want to keep some of the settings that is stored.

Reviewer: Ok, so maybe write out "emergeny shutdown" and clarify that settings may be lost.

Author: done

```

> ## Production settings(default)

Reviewer: ```suggestion
## Production settings (default)
```

> - Sim card detect 

Reviewer: ```suggestion
- Detect SIM card
```

  - AT+QSIMDET=0,0

>     - Need restart to take affect

Reviewer: ```suggestion
    - Needs a restart to take affect
```

```
echo 1 > /opt/hm/pic_attributes/ctrl_modem_on
```
* The modem takes about 10 seconds to start and register itself with the kernel. It then takes up to one minute before it is registered in the linux usb list.

* Check USB TTY device to see if it has shown up.

```bash
find /dev -name "ttyUSB*"
```

```bash
/dev/ttyUSB3
/dev/ttyUSB2
/dev/ttyUSB1
/dev/ttyUSB0
```

* The modem also shows up with the `lsusb` command.

```bash
root@10210100023:~# lsusb | grep Quectel
Bus 001 Device 007: ID 2c7c:030a Quectel Wireless Solutions Co., Ltd. Quectel EM05-G
```

# Status ON

Example showing a modem start and running.

```bash
/opt/hm/modem_status.sh
******* MODEM STATUS DUMP *******

ctrl_modem_status: 1342177491

Modem state     ON
Ctrl state:     ON

SLED:           0
SLED Mon:       ON

Modem EMG/reset:        0

Current state:  on
Prev state:     turn_on_timeout

Prev RC:        unexpected_shutdown

Start cnt:      0
Reset cnt:      0

EMG used:       NO
```

## Shutdown/Reset 
To disable the modem, set ctrl_modem_on to 0

To reset it use ctrl_modem_emg (emergency restart signal) and set it to 0, then back to 1 after a second.

```bash
echo 0 > /opt/hm/pic_attributes/ctrl_modem_on

> # or (not to be used regulary because it could effect the memory)

Reviewer: ```suggestion
# or (not to be used regularly because it could affect the memory)
```
